x-pack/plugins/apm/public/components/shared/FilterBar/DatePicker.tsx

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import datemath from '@elastic/datemath';
+import { EuiSuperDatePicker, EuiSuperDatePickerProps } from '@elastic/eui';
+import { memoize } from 'lodash';
+import React from 'react';
+import { connect } from 'react-redux';
+import { RouteComponentProps, withRouter } from 'react-router-dom';
+import {
+  toBoolean,
+  toNumber,
+  updateTimePicker
+} from 'x-pack/plugins/apm/public/store/urlParams';
+import { fromQuery, toQuery } from '../Links/url_helpers';
+
+interface Props extends RouteComponentProps {
+  dispatchUpdateTimePicker: typeof updateTimePicker;
+}
+
+interface DatePickerParams {
+  rangeFrom: string;
+  rangeTo: string;
+  refreshPaused: boolean;
+  refreshInterval: number;
+}
+
+const APM_DEFAULT_TIME_OPTIONS: DatePickerParams = {
+  rangeFrom: 'now-24h',
+  rangeTo: 'now',
+  refreshPaused: true,
+  refreshInterval: 0
+};
+
+class DatePickerComponent extends React.Component<Props> {
+  public refreshTimeoutId = 0;
+
+  public getParamsFromSearch = memoize((search: string) => {
+    const query = toQuery(search);
+    if ('refreshPaused' in query) {
+      query.refreshPaused = toBoolean(query.refreshPaused);
+    }
+    if ('refreshInterval' in query) {
+      query.refreshInterval = toNumber(query.refreshInterval as string);
+    }
+    return {
+      ...APM_DEFAULT_TIME_OPTIONS,
+      ...query
+    } as DatePickerParams;
+  });
```

Comment:
I had a little bit of a hard time, understanding what this method was actually returning due to the conditionals and spreads. Since it's only four params I think it's worth making them explicit. What do you think about:

```ts
public getParamsFromSearch = (search: string) => {
  const { rangeFrom, rangeTo, refreshPaused, refreshInterval } = toQuery(
    search
  );
  return {
    rangeFrom: rangeFrom || 'now-24h',
    rangeTo: rangeTo || 'now',
    refreshPaused: toBoolean(refreshPaused),
    refreshInterval: toNumber(refreshInterval) || 0
  };
};
```

btw. This requires some minor modifications to `APMQueryParams` in url_helpers

Reply:
Hm, yeah, that function could use some work. This suggestion makes it much harder to see the defaults at a glance, and actually prevents us from being able to use a default value for `refreshPaused` which we want to default to `true`. Once I get the types sorted based on your good feedback there, I'll look at this again and see what we can do.

Reply:
ah, missed the default for `refreshPaused`. Maybe `toBoolean` should take a default value as a second param?

Reply:
OK I was actually able to move _all_ of these defaults up into the destructure, which reads nicely and avoids the confusion from before.

Reply:
Looks good 👍